crates/llm-local/src/token_output_stream.rs

```diff
@@ -0,0 +1,68 @@
+/// Implementation for TokenOutputStream Code is borrowed from
+/// https://github.com/huggingface/candle/blob/main/candle-examples/src/token_output_stream.rs
+/// (Commit SHA 4fd00b890036ef67391a9cc03f896247d0a75711)
+///
+/// This is a wrapper around a tokenizer to ensure that tokens can be returned to the user in a
```

Comment:
Nit: typically doc comments start with a description of the type and then add other notes (such as where the code may be borrowed from) at the end. This seems to be the other way around.